src/main/java/org/folio/processing/events/services/publisher/RestEventPublisher.java

```diff
@@ -14,32 +13,19 @@
 import java.util.concurrent.CompletableFuture;
 
 public class RestEventPublisher implements EventPublisher {
+
   private static final Logger LOGGER = LoggerFactory.getLogger(RestEventPublisher.class);
 
   @Override
   public CompletableFuture<Event> publish(DataImportEventPayload eventPayload) {
-    OkapiConnectionParams params = new OkapiConnectionParams()
-      .withOkapiUrl(eventPayload.getOkapiUrl())
-      .withTenantId(eventPayload.getTenant())
-      .withToken(eventPayload.getToken());
-    Event event = new Event()
-      .withId(UUID.randomUUID().toString())
-      .withEventType(eventPayload.getEventType())
-      .withEventPayload(JsonObject.mapFrom(eventPayload).encode());
-    return postPubsubPublish(event, params);
-  }
-
-  /**
-   * Sends event to mod-pubsub using REST client
-   *
-   * @param event  event
-   * @param params connection parameters
-   * @return future
-   */
-  private CompletableFuture<Event> postPubsubPublish(Event event, OkapiConnectionParams params) {
     CompletableFuture<Event> future = new CompletableFuture<>();
-    PubsubClient client = new PubsubClient(params.getOkapiUrl(), params.getTenantId(), params.getToken());
     try {
+      Event event = new Event()
+        .withId(UUID.randomUUID().toString())
+        .withEventType(eventPayload.getEventType())
+        .withEventPayload(JsonObject.mapFrom(eventPayload).encode());
+
+      PubsubClient client = new PubsubClient(eventPayload.getOkapiUrl(), eventPayload.getTenant(), eventPayload.getToken());
       client.postPubsubPublish(event, response -> {
```

Comment:
Could you please descirbe - why did we using "postPubsubPublish"-method, instead "sendEventMessage" from  PubSubClientUtils?

Reply:
it was written when PubSubClientUtils didn't exist yet. will update, thanks